examples/pennylane_universal_classifier.py

```diff
@@ -0,0 +1,491 @@
+outside a circle. The goal is to train a quantum circuit to predict the
+label (red or blue) given an input point’s coordinate.
+
+.. figure:: ../../examples/figures/universal_circles.png
+   :scale: 65%
+   :alt: circles
+
+
+Transforming quantum states using unitary operations
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A single-qubit quantum state is characterized by a two-dimensional state
+vector and can be visualized as a point in the so-called Bloch sphere.
+Instead of just being a 0 (up) or 1 (down), it can exist in a
+superposition with say 30% chance of being in the :math:`|0 \rangle` and
+70% chance of being in the :math:`|1 \rangle` state. This is represented
+by a state vector :math:`\psi \rangle = 0.3|0 \rangle + 0.7|1 \rangle ` - the probability
+“amplitude” of the quantum state. In general we can take a vector
+:math:`(\alpha, \beta)` to represent the probabilities of a qubit being
+in a particular state and visualize it on the Bloch sphere as an arrow.
+
+.. figure:: ../../examples/figures/universal_bloch.png
+   :scale: 65%
+   :alt: bloch
+
+Data loading using unitaries
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In order to load data onto a single qubit, we use a unitary operation U
+(:math:`x_1`, :math:`x_2`, :math:`x_3`) which is just a parameterized
+matrix multiplication representing the rotation of the state vector in
+the Bloch sphere. Eg., to load :math:`(x_1, x_2)` into the qubit, we
+just start from some initial state vector, :math:`|0 \rangle `,
+apply the unitary operation :math:`U(x_1, x_2, 0)` and end up at a new
+point on the Bloch sphere. Here we have padded 0 since our data is only
+2D. Adri´an P´erez-Salinas et al. (2019) discuss how to load a higher
+dimensional data point (:math:`[x_1, x_2, x_3, x_4, x_5, x_6]`) by
+breaking it down in sets of three parameters
+(:math:`U(x_1, x_2, x_3), U(x_4, x_5, x_6)`).
+
+Model parameters with data re-uploading
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Once we load the data onto the quantum circuit, we want to have some
+trainable nonlinear model similar to a Neural Network and a way of
+learning the weights of the model from data. This is again done with
+unitaries, :math:`U(\theta_1, \theta_2, \theta_3)` such that we load the
+data first and then apply the weights to form a single layer
+:math:`L(\vec \theta, \vec x) = U(\vec \theta)U(\vec x)`. In principle,
+this is just application of two matrix multiplications on an input
+vector initialized to some value. In order to increase the number of
+trainable parameters (similar to increasing neurons in a single layer of
+a neural network), we can reapply this layer again and again with new
+sets of weights,
+:math:`L(\vec \theta_1, \vec x) L(\vec \theta_2, , \vec x) ... L(\vec \theta_L, \vec x)`
+for :math:`L` layers. The quantum circuit would look like the following:
+
+.. figure:: ../../examples/figures/universal_layers.png
+   :scale: 75%
+   :alt: Layers
```

Comment:
The style of these images is great!

For reference:

![image](https://user-images.githubusercontent.com/2959003/61889958-07c72880-af39-11e9-9df1-7a4bfe70f8d4.png)